lib/thread.c

```diff
@@ -1238,12 +1258,26 @@ static struct thread *thread_run(struct thread_master *m, struct thread *thread,
 }
 
 static int thread_process_io_helper(struct thread_master *m,
-				    struct thread *thread, short state, int pos)
+				    struct thread *thread, short state,
+				    short actual_state, int pos)
 {
 	struct thread **thread_array;
 
-	if (!thread)
+	/*
+	 * If another pthread scheduled this file descriptor for this event
+	 * we're responding to, no problem, we're getting to it now.
+	 * Additionally if !thread if we don't clear this now we'll
+	 * infinaloop( which sucks )
```

Comment:
Probably some better explanation is in order here. The original comment and this one isn't clear on what's going on. Something like,
```
poll() clears the .events field, but the pollfd array we passed to poll()
is a copy of the one used to schedule threads. We need to synchronize
state between the two here by applying the same changes poll() made on the
copy to the "real" pollfd array.
```

Reply:
fixed